### ISSUE TITLE
Maybe write your own code?

### ISSUE THREAD

**Issue by fchollet** [hycis/Mozi#1]
It would probably be a good exercise. Just saying.

**Comment by hycis** [comment]
Most of the code was written in 2012 before keras is out, but you are right, I'm borrowing some of the keras latest implementation like batch normalisation which maybe if i have time, I will try to reimplement it.
